pandas/core/window/ewm.py

```diff
@@ -391,7 +391,9 @@ def __init__(
                 raise ValueError(
                     "halflife must be a string or datetime.timedelta object"
                 )
-            if isna(self.times).any():
+            # error: Item "bool" of "Union[bool, ndarray[Any, dtype[bool_]], NDFrame]"
+            # has no attribute "any"
+            if isna(self.times).any():  # type: ignore[union-attr]
```

Comment:
shouldn't we cast(bool, ...) to avoid this?

Reply:
mypy thinks that `self.times` can still be str at this point - it can't be, is converted [here](https://github.com/pandas-dev/pandas/blob/bb9a9852265915a4688f772dd062d3fcf4159a32/pandas/core/window/ewm.py#L383) to a Series(?). If we decide against the fallback-overload, I will try to avoid this ignore comment.

With the "fallback-overload" we have a different issue (even when self.times is not str): NDFrame and np.ndarray are not covered by the same overload, which makes mypy/pyright pick the (unsupported) fallback overload.

Reply:
I tried this and it worked.  Not sure if we should go this direction.  In `missing.py`, add an explicit overload for the `Union` (right before the last one):
```python
@overload
def isna(obj: NDFrameT | np.ndarray) -> npt.NDArray[np.bool_] | NDFrameT:  # type: ignore[misc]
    ...
```
Then in the code in `ewm.py`, do this:
```python
            assert not isinstance(self.times, str)
            if isna(self.times).any():
                raise ValueError("Cannot convert NaT values to integer")
```
```

Reply:
I added a cast and a comment